src/energyhub.py

```diff
@@ -108,13 +108,22 @@ def _perform_preprocessing_checks(self):
                 for node in topology["nodes"]:
                     for tec_name in self.data.technology_data[period][node]:
                         tec = self.data.technology_data[period][node][tec_name]
-                        if ("ramping_time" in tec.processed_coeff.dynamics) and (
-                            tec.processed_coeff.dynamics["ramping_time"] != -1
+                        if (
```

Comment:
This looks not correct to me:
- if TSA is used or not is not determined by method but by N>0.
- the exception is raised stating method 2, but it is raised for both 1 and 2

Reply:
yes you are right, its fixed now!